application/LicenseAppService.go

```diff
@@ -144,27 +145,42 @@ func (s *LicenseAppService) ModifySeats(req ModifySeatAssignmentRequest) error {
 }
 
 // ProcessOrgEntitledEvent handles the OrgEntitledEvent by storing the license and importing users
-func (s *LicenseAppService) ProcessOrgEntitledEvent(evt OrgEntitledEvent) error {
-	// first, check if there's already a license/org to skip userImport after applying the new license, if necessary.
-	orgIsImported, err := s.seatRepo.IsImported(evt.OrgID)
+func (s *LicenseAppService) ProcessOrgEntitledEvent(evt OrgEntitledEvent, strictMode bool) error {
+	// first, check if there's already a license/org and/or existing org users.
+	licenseImported, usersImported, err := s.seatRepo.IsImported(evt.OrgID, evt.ServiceID)
 	if err != nil {
 		return err
 	}
 
-	err = s.seatRepo.ApplyLicense(&domain.License{
-		OrgID:     evt.OrgID,
-		ServiceID: evt.ServiceID,
-		MaxSeats:  evt.MaxSeats,
-		Version:   "",
-		InUse:     0,
-	})
+	if licenseImported {
+		licExistsErr := fmt.Errorf("License already exists for the given org in %v: ", evt)
 
-	if err != nil {
-		return err
+		if strictMode {
+			return licExistsErr
+		}
+		glog.Warning(licExistsErr)
+
+	} else {
+		// we only create/touch an existing license for a given org if it doesn't already exist
+		err = s.seatRepo.ApplyLicense(&domain.License{
```

Comment:
Hum, wouldn't that mean that you could now only add one license per org? so e.g. adding foobar and barbar for one org is not possible anymore, because there would already be a license?!

Reply:
Forget it, got the logic wrong. In that case licenseExists is false.